### ISSUE TITLE
Freezing when buying all upgrades

### ISSUE THREAD

**Issue by koenvanzuijlen** [Aktanusa/CookieMonster#501]
**Describe the bug**
I was going for 1000 ascensions so using Buy All upgrades a lot, but once in ~20 ascensions it freezes the game. When I click the buy all upgrades button I get the below error:

This started in 2.031.3

**Screenshots**
```
Uncaught TypeError: CM.Cache.Upgrades[Game.UpgradesInStore[CM.Disp.tooltipName].name] is undefined
    UpdateTooltipUpgrade https://aktanusa.github.io/CookieMonster/CookieMonster.js:2698
    UpdateTooltip https://aktanusa.github.io/CookieMonster/CookieMonster.js:2615
    Tooltip https://aktanusa.github.io/CookieMonster/CookieMonster.js:2473
    <anonymous> https://aktanusa.github.io/CookieMonster/CookieMonster.js line 2385 > eval:1
    drawMod https://aktanusa.github.io/CookieMonster/CookieMonster.js line 4036 > eval:17
    draw https://aktanusa.github.io/CookieMonster/CookieMonster.js:4038
    <anonymous> https://aktanusa.github.io/CookieMonster/CookieMonster.js line 2385 > eval:1
CookieMonster.js:2698:2
    UpdateTooltipUpgrade https://aktanusa.github.io/CookieMonster/CookieMonster.js:2698
    UpdateTooltip https://aktanusa.github.io/CookieMonster/CookieMonster.js:2615
    Tooltip https://aktanusa.github.io/CookieMonster/CookieMonster.js:2473
    <anonymous> https://aktanusa.github.io/CookieMonster/CookieMonster.js line 2385 > eval:1
    drawMod https://aktanusa.github.io/CookieMonster/CookieMonster.js line 4036 > eval:17
    draw https://aktanusa.github.io/CookieMonster/CookieMonster.js:4038
    <anonymous> https://aktanusa.github.io/CookieMonster/CookieMonster.js line 2385 > eval:1
```

**Save file**
I don't know if this will help:
```
Mi4wMzF8fDE2MTE1ODQ0MDM0NzE7MTYwOTc0NDg0NjE0NjsxNjExNTg0NDEzNDIyO0pvZXk7c2J6ZGd8MTExMTExMDExMDAxMTExMTAxMDEwfDMuNjQ3NTg1MDY5MTQ0MTcyNWUrMzE7OS43MjY1NjQyNTk2MjM4MjRlKzMyOzA7MjM4MTswOzE4NDswOzA7OC4wNTk3ODg1OTU4NTg0ODhlKzQ5OzA7MDswOzY1OzU0MDA7MjkxOzA7MDswOzA7MDswOzA7OzA7MDs0MzE5NTc2OTY4Mjk2OzIwODIzMjkxOTIyMDQ7MjIzNzI0Nzc3NjA5MjswOzA7NjEzOzQ0Mjs0OTQ7NDI1OzMyMjswOzA7MDswOzUwOzA7MDs1OzIzOzE2MTE1Mjc1NzY1MTA7MDswOzIyNzs0MTswOzA7NS43NDQ0NTM4NzEyMzc1NjRlKzMyO3wyNTcsMjU3LDkuMDEzNzc0NjQ5OTQ5MjQ5ZSsyNywwLCwwLDI1NzsyNTcsMjU3LDYuODcwODE5MTc1Mjc5MjQ1ZSsyNCwwLCwxLDI1NzswLDAsMCw1LDE2MTE1ODQ3MDM1MTQ6MDoxNjExNTg0NDAzNDc2OjA6MDoyMjI6MDowOjE2MTE1ODQ0MDM0NzY6IDExMTEwMDAwMTEwMTExMDAwMDAwMDAwMTExMDAxMDAwMDAgMDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6MDowOjA6LDAsMDswLDAsMCwwLCwxLDA7MCwwLDAsMCwsMSwwOzAsMCwwLDEsMDowOjE6MDoxOiAxMTAwOjE6MTI6NTc5OjA6MTowITIxMDQ6MToyMjo1ODA6MDoxOjAhMzA1NTo1OjcxOjUwOTowOjE6MCE0Mjg0OjE6MzY6MzgwOjA6MTowITUxNzI6NTo4ODo1NjM6MDoxOjAhNjMwNzoxOjI5OjI1NTowOjE6MCE2ODI1OjI6LTEyOjk1NTowOjE6MCE4NjQyOjM6ODU6MTY1OjA6MTowITkxODU6MToxNjo1NzowOjE6MCE5OTc5OjE6MTE6ODQzOjA6MTowITExMDI2OjE6MTo4NTY6MDoxOjAhMTE2MTk6MjotMjg6NjU5OjA6MTowITEyOTAzOjI6LTE0OjYwNzowOjE6MCExNDE0MzowOi0yOjIzMDowOjA6MCExNTI2MTo1Oi0yNjozODg6MDoxOjAhMTY4NzU6NDotODE6NDc2OjA6MDowISAwLDAsMDswLDAsMCwxLC0xLy0xLy0xIDMgMTYxMTU4NDQwMzQ3OCAwLDAsMDswLDAsMCwxLDUgMCA1NSAwLDAsMDswLDAsMCwwLCwxLDA7MCwwLDAsMCwsMSwwOzAsMCwwLDAsLDEsMDswLDAsMCwwLCwxLDA7MCwwLDAsMCwsMSwwOzAsMCwwLDAsLDEsMDswLDAsMCwwLCwxLDA7MjA0LDIwNCwzLjYwNTgwNTQxMDExNTI4NDNlKzMxLDAsLDEsMjA0OzAsMCwwLDAsLDEsMDsxNjAsMTYwLDkuMzY1ODkzNTEyMTU3NjA0ZSszMiwwLCwxLDE2MDt8MTExMTExMTExMTExMTExMTExMTEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMDAwMDAwMDAwMDAwMDAwMDAwMTExMTExMDAwMDAwMDAwMDAwMDAxMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTExMTAwMDAwMDAwMDAxMTExMTEwMDExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDExMTExMTExMTEwMDExMDAwMDAwMDAwMDAwMDAwMDAwMTExMTExMTEwMDExMTExMTExMTExMTExMDAwMDAwMDAwMDAwMDAwMDAwMTEwMDAwMDAwMDAwMDAwMDAwMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTEwMTAxMDEwMDAxMTAwMDAwMDAwMTEwMDAwMDAwMDAwMDAwMDAwMDAxMTExMTExMTExMTEwMDEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTExMTExMTExMTExMTExMTExMDAxMDAwMTAwMDAwMDAwMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDExMTExMTExMDAwMDExMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMTExMTExMTExMTAwMDAwMDExMTExMTExMTExMTExMTExMTExMTExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTAwMTExMTExMTExMTExMTExMTEwMTExMTExMTExMTExMDAwMDAwMTAxMTExMTExMDEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTExMTAwMDAwMDAwMDAxMTExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTExMTExMTExMTExMTExMTEwMDAwMDAwMDAwMTEwMDAwMTEwMDExMTExMTExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMDAwMDAwMDAwMTExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTExMTExMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMTExMTAwMDAwMDAwMTExMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAxMTAwMDAxMTExMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDExMTExMTExMTExMTExMDAwMDAwMDAwMDAwMDAxMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMTExMTExMTEwMDAwMDAwMDExMDAxMTExfDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTExMTExMTAxMTExMTEwMTExMTExMDAwMDAwMDAxMTExMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTExMTAwMDEwMDExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMDAwMDAwMDAwMDAwMDEwMDAxMTExMTExMTExMTAxMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTExMTExMTExMTEwMDAwMTExMTExMTExMTEwMDEwMTExMTExMTExMTExMTExMTExMTExMTExMTExMTEwMTAxMTExMTExMTExMTExMDExMTAxMTEwMDAwMDAxMDAwMDAwMDEwMDAwMDAwMTExMTExMTExMTExMTAwMDAwMDEwMTExMTExMDAwMDAwMDAwMTExMTExMTExMDAwMDExMTAxMTExMDAwMDAwMDB8fENvb2tpZU1vbnN0ZXI6eyJzZXR0aW5ncyI6eyJCb3RCYXIiOjEsIlRpbWVyQmFyIjoxLCJUaW1lckJhclBvcyI6MCwiQnVpbGRDb2xvciI6MSwiQnVsa0J1aWxkQ29sb3IiOjAsIkNvbG9yUFBCdWxrTW9kZSI6MCwiVXBCYXJDb2xvciI6MSwiVXBncmFkZUJhckZpeGVkUG9zIjoxLCJDYWxjV3JpbmsiOjAsIkNQU01vZGUiOjEsIkF2Z0NQU0hpc3QiOjMsIkF2Z0NsaWNrc0hpc3QiOjAsIlRvb2xXYXJuQm9uIjowLCJHQ05vdGlmaWNhdGlvbiI6MCwiR0NGbGFzaCI6MSwiR0NTb3VuZCI6MSwiR0NWb2x1bWUiOjEwMCwiR0NTb3VuZFVSTCI6Imh0dHBzOi8vZnJlZXNvdW5kLm9yZy9kYXRhL3ByZXZpZXdzLzY2LzY2NzE3XzkzMTY1NS1scS5tcDMiLCJGb3J0dW5lTm90aWZpY2F0aW9uIjowLCJGb3J0dW5lRmxhc2giOjEsIkZvcnR1bmVTb3VuZCI6MSwiRm9ydHVuZVZvbHVtZSI6MTAwLCJGb3J0dW5lU291bmRVUkwiOiJodHRwczovL2ZyZWVzb3VuZC5vcmcvZGF0YS9wcmV2aWV3cy8xNzQvMTc0MDI3XzMyNDI0OTQtbHEubXAzIiwiU2VhTm90aWZpY2F0aW9uIjowLCJTZWFGbGFzaCI6MSwiU2VhU291bmQiOjEsIlNlYVZvbHVtZSI6MTAwLCJTZWFTb3VuZFVSTCI6Imh0dHBzOi8vd3d3LmZyZWVzb3VuZC5vcmcvZGF0YS9wcmV2aWV3cy8xMjEvMTIxMDk5XzIxOTMyNjYtbHEubXAzIiwiR2FyZEZsYXNoIjoxLCJHYXJkU291bmQiOjEsIkdhcmRWb2x1bWUiOjEwMCwiR2FyZFNvdW5kVVJMIjoiaHR0cHM6Ly9mcmVlc291bmQub3JnL2RhdGEvcHJldmlld3MvMTAzLzEwMzA0Nl84NjE3MTQtbHEubXAzIiwiTWFnaWNOb3RpZmljYXRpb24iOjAsIk1hZ2ljRmxhc2giOjEsIk1hZ2ljU291bmQiOjEsIk1hZ2ljVm9sdW1lIjoxMDAsIk1hZ2ljU291bmRVUkwiOiJodHRwczovL2ZyZWVzb3VuZC5vcmcvZGF0YS9wcmV2aWV3cy8yMjEvMjIxNjgzXzEwMTUyNDAtbHEubXAzIiwiV3JpbmtsZXJOb3RpZmljYXRpb24iOjAsIldyaW5rbGVyRmxhc2giOjEsIldyaW5rbGVyU291bmQiOjEsIldyaW5rbGVyVm9sdW1lIjoxMDAsIldyaW5rbGVyU291bmRVUkwiOiJodHRwczovL2ZyZWVzb3VuZC5vcmcvZGF0YS9wcmV2aWV3cy8xMjQvMTI0MTg2XzgwNDMtbHEubXAzIiwiV3JpbmtsZXJNYXhOb3RpZmljYXRpb24iOjAsIldyaW5rbGVyTWF4Rmxhc2giOjEsIldyaW5rbGVyTWF4U291bmQiOjEsIldyaW5rbGVyTWF4Vm9sdW1lIjoxMDAsIldyaW5rbGVyTWF4U291bmRVUkwiOiJodHRwczovL2ZyZWVzb3VuZC5vcmcvZGF0YS9wcmV2aWV3cy8xNTIvMTUyNzQzXzE1NjYzLWxxLm1wMyIsIlRpdGxlIjoxLCJUb29sdGlwQnVpbGRVcCI6MSwiVG9vbHRpcEFtb3IiOjAsIlRvb2xXYXJuTHVja3kiOjEsIlRvb2xXYXJuTHVja3lGcmVuenkiOjEsIlRvb2xXYXJuQ29uanVyZSI6MSwiVG9vbFdhcm5FZGlmaWNlIjoxLCJUb29sV2FyblBvcyI6MSwiVG9vbHRpcEdyaW0iOjEsIlRvb2xXcmluayI6MSwiVG9vbHRpcEx1bXAiOjEsIkRyYWdvbkF1cmFJbmZvIjoxLCJTdGF0cyI6MSwiTWlzc2luZ1VwZ3JhZGVzIjoxLCJVcFN0YXRzIjoxLCJUaW1lRm9ybWF0IjowLCJEZXRhaWxlZFRpbWUiOjEsIkdyaW1vaXJlQmFyIjoxLCJTY2FsZSI6MiwiU2NhbGVEZWNpbWFscyI6MiwiU2NhbGVTZXBhcmF0b3IiOjAsIlNjYWxlQ3V0b2ZmIjo5OTk5OTksIkNvbG9ycyI6eyJCbHVlIjoiIzRiYjhmMCIsIkdyZWVuIjoiIzAwZmYwMCIsIlllbGxvdyI6IiNmZmZmMDAiLCJPcmFuZ2UiOiIjZmY3ZjAwIiwiUmVkIjoiI2ZmMDAwMCIsIlB1cnBsZSI6IiNmZjAwZmYiLCJHcmF5IjoiI2IzYjNiMyIsIlBpbmsiOiIjZmYxNDkzIiwiQnJvd24iOiIjOGI0NTEzIn0sIlNvcnRCdWlsZGluZ3MiOjAsIlNvcnRVcGdyYWRlcyI6MCwiR0NUaW1lciI6MSwiRmF2aWNvbiI6MSwiV3JpbmtsZXJCdXR0b25zIjoxLCJIZWFkZXIiOnsiQmFyc0NvbG9ycyI6MSwiQ2FsY3VsYXRpb24iOjEsIk5vdGlmaWNhdGlvbiI6MSwiTm90aWZpY2F0aW9uR0MiOjEsIk5vdGlmaWNhdGlvbkZDIjoxLCJOb3RpZmljYXRpb25TZWEiOjEsIk5vdGlmaWNhdGlvbkdhcmQiOjEsIk5vdGlmaWNhdGlvbk1hZ2kiOjEsIk5vdGlmaWNhdGlvbldyaW5rIjoxLCJOb3RpZmljYXRpb25Xcmlua01heCI6MSwiVG9vbHRpcCI6MSwiU3RhdGlzdGljcyI6MSwiTm90YXRpb24iOjEsIk1pc2NlbGxhbmVvdXMiOjEsIkx1Y2t5IjoxLCJTcGVsbHMiOjEsIkNoYWluIjoxLCJQcmVzdGlnZSI6MSwiV3JpbmsiOjEsIlNlYSI6MSwiTWlzYyI6MX19LCJ2ZXJzaW9uIjoiMi4wMzEuMyJ9Ow%3D%3D%21END%21
```

**Comment by DanielNoord** [comment]
Likely that during reloading after the ascension one of the Cache functions is not immediately called and you hover over the "buy all upgrades" button before it does.
Think we should be able to fix this be redoing all Caches immediately after ascending.
However, I have not had time to actually check the code to see if this might be the problem.

**Comment by DanielNoord** [comment]
I think this has been solved with #572. I am closing this for now, but please get back to me if the problems persist!
